### PR TITLE
Upgrade scylla rust driver to v0.9

### PR DESCRIPTION
**Pull request by cleaton**

### CODE REVIEW

**Comment by cleaton** [comment]
I've added additional benchmark pure throughput benchmark using Task.async_stream instead of relying on benchee. After some digging I found ColSpec returned in QueryResult is very expensive. After removing ColSpec from result exscylla is consistently faster than erlcass using Task.async_stream, no matter the concurrency. On my machine, concurrency 500:
```
erlcass: ~36000 query/s
ex_scylla: ~39000 query/s
```
With ColSpec i get ~26000 query/s.

The benchee setup still shows ex_scylla is ~1.1 - 1.3 slower than erlcass but I am starting to doubt how valid it is to use high parallelism in benchee tests.

The branch is not complete yet. Functions to set execution context are not fully there and I didn't do any tuning on the scylla session such as enabling `write_coalescing`.

**Comment by cleaton** [comment]
Actually running the benchee benchmarks I can see a big network difference, 23MiB/s for erlcass while ex_scylla is using 60MiB/s! enabling `write_coalescing` did not make a difference.

File: native/ex_scylla/.cargo/config
Comment (louiscb) [inline native/ex_scylla/.cargo/config:1]: I had to set the config to this to get the project working on my M1 Mac. I copied it from Rustlers example project
https://github.com/rusterlium/NifIo/blob/master/native/io/.cargo/config

```
[target.x86_64-apple-darwin]
rustflags = [
    "-C", "link-arg=-undefined",
    "-C", "link-arg=dynamic_lookup",
]

[target.aarch64-apple-darwin]
rustflags = [
    "-C", "link-arg=-undefined",
    "-C", "link-arg=dynamic_lookup",
]

# See https://github.com/rust-lang/rust/issues/59302
[target.x86_64-unknown-linux-musl]
rustflags = [
  "-C", "target-feature=-crt-static"
]
```

**Comment by cleaton** [comment]
I asked in the rust-driver slack about the network situation and it is most likely due to this: https://github.com/scylladb/scylla-rust-driver/issues/169 

I've also realized the way the current rust -> elixir is quite inefficient. There are two steps:
1. Recursively convert scylla type into the scyllawrapper type which support Encode<> trait
2. Call encode, which recursively calls encode on all nested types.

Preferably there should be a way to encode everything in a single pass instead of going through all items twice, especially for hot items such as `QueryResult`.

For now I think i'll just work on adding the missing API calls from existing to make 0.9 work fully. I'll review the encoding again once https://github.com/scylladb/scylla-rust-driver/pull/665 is merged as it might cause a large change.

**Pull request merged**